httpx/_content.py

```diff
@@ -52,9 +52,7 @@ def __iter__(self) -> Iterator[bytes]:
             raise StreamConsumed()
 
         self._is_stream_consumed = True
-        if hasattr(self._stream, "read") and not isinstance(
-            self._stream, SyncByteStream
-        ):
```

Comment:
The `and not isinstance(self._stream, SyncByteStream)` is what prompted me to clean these up.

It's such an oddly confusing branch - I couldn't figure out why it existed.

*"Treat file-like objects as file-like objects, except for this one specific case"*.

No thanks.